ec2utils/ec2uploadimg/lib/ec2utils/ec2uploadimg.py

```diff
@@ -376,8 +376,10 @@ def _create_target_root_volume(self):
     def _create_upload_key_pair(self, key_name='temporary_ec2_uploadkey'):
         if self.verbose:
             print('Creating temporary key pair')
-        home_dir = os.path.expanduser('~/')
-        fd, location = mkstemp(prefix='temporary_ec2_uploadkey.', suffix='.key', dir=home_dir)
+        dir_path = os.path.expanduser('~/')
+        if not os.access(dir_path, os.W_OK):
+            dir_path = '/tmp/'
```

Comment:
I am not OK with /tmp, I understand the need, use mkdtemp()

Reply:
@rjschwei ok